src/base/utils/foreignapps.cpp

```diff
@@ -69,26 +77,181 @@ namespace
                 return false;
             }
 
-            LogMsg(QCoreApplication::translate("Utils::ForeignApps", "Python detected, version: %1").arg(info.version), Log::INFO);
+            LogMsg(QCoreApplication::translate("Utils::ForeignApps", "Python detected, executable name: '%1', version: %2")
+                .arg(info.executableName, info.version), Log::INFO);
             return true;
         }
 
         return false;
     }
+
+#if defined(Q_OS_WIN)
+    enum REG_SEARCH_TYPE
```

Comment:
Is this part of code moved w/o changes?

Reply:
yes.

Reply:
>yes, if you mean the enum.

No, I mean all python searching code moved from misc.cpp
>The minor changes are at

IIRC, the second change is "return python exe path instead of python exe folder path"?

Reply:
>IIRC, the second change is "return python exe path instead of python exe folder path"?

Yes.